### PR TITLE
File based dev registry

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

Fixes https://github.com/cloudflare/workers-sdk/issues/6037

## Author has addressed the following

- Tests
  - [x] TODO (before merge)
  - [ ] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: not public facing—internal stability refactor gated behind a feature flag

### CODE REVIEW

**Comment by penalosa** [comment]
- Add file heartbeat to detect crashed workers
- Add Wrangler version to files to detect back compat
- Add cleanup mechanism when Wrangler tries to fetch a worker that no longer exists

File: packages/wrangler/src/dev.tsx [hunk 1]
```diff
@@ -295,6 +296,13 @@ export function devOptions(yargs: CommonYargsArgv) {
 					"Use the experimental DevEnv instantiation (unified across wrangler dev and unstable_dev)",
 				default: false,
 			})
+			.option("experimental-registry", {
+				alias: ["x-registry"],
```
Comment (RamIdeas) [inline packages/wrangler/src/dev.tsx:300]: 👏

File: packages/wrangler/src/experimental-flags.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+const flags = new AsyncLocalStorage<ExperimentalFlags>();
+
+export const run = <V>(flagValues: ExperimentalFlags, cb: () => V) =>
+	flags.run(flagValues, cb);
+
+export const FILE_BASED_REGISTRY = () => flags.getStore()?.FILE_BASED_REGISTRY;
+
+// TODO: actually use this
+export const DEV_ENV = () => flags.getStore()?.DEV_ENV;
```
Comment (RamIdeas) [inline packages/wrangler/src/experimental-flags.ts:16]: Can we discuss alternatives to this interface please?

Comment (RamIdeas) [reply packages/wrangler/src/experimental-flags.ts:16]: Minor point about interface:

I'd like to make adding a feature flag as easy as adding it to a type with no repetition. Having an exported function per flag requires multiple changes to this file to add each flag. While this allows us to find all usages of _each_ feature flag with Find All References, they won't be easy to find usages of _any_ flag. Can we try something like `export const XFLAG = <F extends keyof ExperimentalFlags>(flag: F) => ExperimentalFlags[F];`?

More for discussion:

This is a great use of AsyncLocalStorage. Can we generalise this use for other singletons like Logger etc? Are there any parts of wrangler this wouldn't work for?

**Review by RamIdeas — APPROVED** [review]

**Review by petebacondarwin — COMMENTED** [review]
Didn't have time to review it. But I see that @RamIdeas has approved.

**Pull request merged**